### ISSUE TITLE
Directives aren't hoisted

### ISSUE THREAD

**Issue by olov** [babel/babel#980]
```javascript
class MyCtrl {
  constructor(a) {
    "any directive prologue";
    foo;
  }
}
```

is incorrectly converted to 
```javascript
"use strict";

var _classCallCheck = function (instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } };

var MyCtrl = function MyCtrl(a) {
  _classCallCheck(this, MyCtrl);

  "any directive prologue";
  foo;
};
```

Note how the directive prologue has become a (pointless) string literal because Babel changed its position. 

BTW thanks for Babel!

(via https://github.com/olov/ng-annotate/issues/64#issuecomment-77865841)

**Comment by olov** [comment]
Wow that's really fast. Did a mini-review on the fix as a thank you. :) https://github.com/babel/babel/commit/672118149a7bd0c927bd00f223ff717fc9e13f14#commitcomment-10101411

**Comment by sebmck** [comment]
Fixed as of 4.7.4, thanks!
